lib/node_modules/@stdlib/array/base/count-if/lib/main.js

```diff
@@ -0,0 +1,155 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isComplexTypedArray = require( '@stdlib/array/base/assert/is-complex-typed-array' );
+var isAccessorArray = require( '@stdlib/array/base/assert/is-accessor-array' );
+var resolveGetter = require( '@stdlib/array/base/resolve-getter' );
+var reinterpret = require( '@stdlib/strided/base/reinterpret-complex' );
+
+
+// FUNCTIONS //
+
+/**
+* Counts the number of elements in an indexed array that satisfy the provided testing function.
+*
+* @private
+* @param {Collection} x - input array
+* @param {(...args: any[]) => boolean} predicate - testing array
+* @param {*} [thisArg] - function context
+* @returns {NonNegativeInteger} number of values for which the provided function evaluates to true
+*
+* @example
+* var x = [ 0, 1, 0, 1 ];
+* var predicate = function( v, thisArg ){ return v > thisArg; }
+* var n = indexed( x, predicate, 0 );
+* // returns 2
+*/
+function indexed( x, predicate, thisArg ) {
+	var n;
+	var i;
+
+	n = 0;
+	for ( i = 0; i < x.length; i++ ) {
+		if ( predicate( x[ i ], thisArg ) ) {
```

Comment:
Please change the arguments passed to the predicate function here and elsewhere to become

```
		if ( predicate( x[ i ], i, x, thisArg ) ) {
```

so that one can use the array index and the input array inside the predicate function, matching other base array functions such as `array/base/none-by` or `array/base/any-by`.

Reply:
I believe the call signature should be

```javascript
if ( predicate.call( thisArg, x[ i ], i, x ) {
```

correct?

Reply:
@kgryte Yes; edited.